lib/redis/cluster.rb

```diff
@@ -0,0 +1,467 @@
+    #   #=> { start_slot: 0,
+    #         end_slot: 12345,
+    #         master: { ip: '127.0.0.1', port: 6379, node_id: '09dbe9720cda62f7865eabc5fd8857c5d2678366' },
+    #         replicas: [{ ip: '127.0.0.1', port: 6380, node_id: '821d8ca00d7ccf931ed3ffc7e3db0599d2271abf' }] }
+    #
+    # @param response [Array<Array>] the raw data of slots info
+    #
+    # @return [Array<Hash>] parsed data
+    def cluster_slots(response)
+      response.map do |res|
+        first_slot, last_slot = res[0..1]
+        master = { ip: res[2][0], port: res[2][1], node_id: res[2][2] }
+        replicas = res[3..-1].map { |r| { ip: r[0], port: r[1], node_id: r[2] } }
+        { start_slot: first_slot, end_slot: last_slot, master: master, replicas: replicas }
+      end
+    end
+
+    # Parse `CLUSTER NODES` command response raw data.
+    #
+    # @example Basic example
+    #   cluster_nodes('07c37dfeb235213a872192d90877d0cd55635b91 127.0.0.1:6379 myself,master - 0 1426238317239 4 connected 0-12345')
+    #   #=> [{ node_id: '07c37dfeb235213a872192d90877d0cd55635b91',
+    #          ip_port: '127.0.0.1:6379',
+    #          flags: ['myself', 'master'],
+    #          master_node_id: '-',
+    #          ping_sent: '0',
+    #          pong_recv: '1426238317239',
+    #          config_epoch: '4',
+    #          link_state: 'connected',
+    #          slots: (0..12345) }]
+    #
+    # @param response [String] the raw data of nodes info
+    #
+    # @return [Array<Hash>] parsed data
+    def cluster_nodes(response)
+      response
+        .split(/[\r\n]+/)
+        .map { |str| deserialize_node_info(str) }
+    end
+
+    # Parse `CLUSTER SLAVES` command response raw data.
+    #
+    # @example Basic example
+    #   cluster_slaves(['07c37dfeb235213a872192d90877d0cd55635b91 127.0.0.1:6379 myself,master - 0 1426238317239 4 connected 0-12345'])
+    #   #=> [{ node_id: '07c37dfeb235213a872192d90877d0cd55635b91',
+    #          ip_port: '127.0.0.1:6379',
+    #          flags: ['myself', 'master'],
+    #          master_node_id: '-',
+    #          ping_sent: '0',
+    #          pong_recv: '1426238317239',
+    #          config_epoch: '4',
+    #          link_state: 'connected',
+    #          slots: (0..12345) }]
+    #
+    # @param response [Array<String>] the raw data of slaves info
+    #
+    # @return [Array<Hash>] parsed data
+    def cluster_slaves(response)
+      response.map { |str| deserialize_node_info(str) }
+    end
+
+    # Parse `CLUSTER INFO` command response raw data.
+    #
+    # @example Basic example
+    #   cluster_info('cluster_state:ok\r\ncluster_size:3')
+    #   #=> { cluster_state: 'ok', cluster_size: '3' }
+    #
+    # @param response [String] the raw data of cluster info
+    #
+    # @return [Hash{Symbol => String}] parsed data
+    def cluster_info(response)
+      response
+        .split(/[\r\n]+/)
+        .map { |str| str.split(':') }
+        .map { |arr| [arr.first.to_sym, arr[1]] }
```

Comment:
What's the rationale for converting to a Symbol?

Reply:
Because I thought that we are use to using Symbol as a Hash key in Ruby. But I don't have rationales. Should we remove it?